src/resources/js/events-dynamic.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Event Date and Time Dynamic Helper Text
+ *
+ * @var object tribe_dynamic_helper_text
+ */
+var tribe_dynamic_helper_text = tribe_dynamic_helper_text || {};
+
+(function ( $, obj ) {
+	'use strict';
+
+	obj.field_class = '.event-dynamic-helper-text';
+	obj.date_fmt_settings = {
+		dateSettings: {
+			days: tribe_dynamic_help_text.days,
+			daysShort: tribe_dynamic_help_text.daysShort,
+			months: tribe_dynamic_help_text.months,
+			monthsShort: tribe_dynamic_help_text.monthsShort,
+		}
+	};
+	obj.date_fmt = new DateFormatter( obj.date_fmt_settings );
+
+	obj.text = JSON.parse( tribe_dynamic_help_text.msgs );
+
+	//Date Formats
+	obj.date_with_year = tribe_dynamic_help_text.date_with_year;
+	obj.date_no_year = tribe_dynamic_help_text.date_no_year;
+	obj.datepicker_format = tribe_dynamic_help_text.datepicker_format;
+
+	//Setup object variables
+	obj.dynamic_text, obj.start_date, obj.start_time, obj.end_date, obj.end_time, obj.all_day = '';
+
+	/**
+	 * Setup Dynamic Text on Load
+	 */
+	obj.init = function () {
+
+		//setup text and display
+		obj.setup_and_display_text();
+
+		//detect event date & time changes
+		obj.event_date_change();
+
+	};
+
+	/**
+	 * Wrapper Method to call methods to update and display text
+	 */
+	obj.setup_and_display_text = function () {
+
+		//get current field values
+		obj.update();
+
+		//determine message to use based on values
+		obj.msg_logic();
+
+		//parse the message and insert into dymanic text div
+		obj.parse_and_display_text();
+
+	};
+
+	/**
+	 * Get Event Date and Time Values
+	 */
+	obj.update = function () {
+
+		obj.start_date = $( '#EventStartDate' ).val();
+		obj.start_time = $( '#EventStartTime' ).val();
+		obj.end_date = $( '#EventEndDate' ).val();
+		obj.end_time = $( '#EventEndTime' ).val();
+		obj.all_day = $( '#allDayCheckbox' ).prop( 'checked' ) ? true : '';
+
+	};
+
+	/**
+	 * Determine Message to Use based on Date and Time
+	 */
+	obj.msg_logic = function () {
+
+		//single date, different start and end time
+		if ( obj.start_date == obj.end_date && !obj.all_day && obj.start_time != obj.end_time ) {
+			obj.dynamic_text = obj.text[0];
+
+			//single date, same start and end time
```

Comment:
The indentation is kind of odd here - I see that the comment is aligned with the inner contents of the `else if() {}` but it probably shouldn't be since we haven't yet entered that block (or, the comment could be moved inside the block).